### PR TITLE
DOC: Further improvements to optimize._lsq docstrings

### PR DESCRIPTION
**Pull request by nmayorov**
I read through the docstrings of public functions again and spotted some mistakes / inconsistencies. 

Sorry, that didn't do it in the previous PR.

### CODE REVIEW

File: scipy/optimize/_lsq/lsq_linear.py [hunk 1]
```diff
@@ -110,9 +110,9 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
     fun : ndarray, shape (m,)
         Vector of residuals at the solution.
     optimality : float
-        First-order optimality measure. The uniform norm of the gradient,
-        scaled as described in Notes if the unconstrained solution is not
-        optimal.
+        First-order optimality measure. It is the quantity related to the
+        gradient of the cost function, which was compared with `tol` during
+        iterations.
```
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: This one is not an improvement. "The [I presume you mean a?] quantity related to the gradient" is less precise than the previous explanation.

File: scipy/optimize/_lsq/lsq_linear.py [hunk 2]
```diff
@@ -142,48 +142,48 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
...
 
     Method 'bvls' runs a Python implementation of the algorithm described in
     [BVLS]_. The algorithm maintains active and free sets of variables, on
     each iteration chooses a new variable to move from the active set to the
-    free set and then solves an unconstrained least-squares problem on free
-    variables. This method gives very accurate solution but may require up to
-    n iterations. Additionally, an ad-hoc initialization procedure is
+    free set and then solves the unconstrained least-squares problem on free
+    variables. This algorithm is guaranteed to give an accurate solution
+    eventually, but may require the number of iterations up to the number
```
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: "may require up to n iterations for a problem with n variables" (?)

File: scipy/optimize/_lsq/lsq_linear.py [hunk 3]
```diff
@@ -142,48 +142,48 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
...
+    of variables. Additionally, an ad-hoc initialization procedure is
     implemented, that determines which variables to set free or active
     initially. It takes some number of iterations before actual BVLS starts,
-    but can significantly reduce the number of iterations required by BVLS.
+    but can significantly reduce the number of future iterations.
```
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: Do you mean "further iterations"?

File: scipy/optimize/_lsq/lsq_linear.py [hunk 4]
```diff
@@ -142,48 +142,48 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
...
+              Minimization Problems," SIAM Journal on Scientific Computing,
+              Vol. 21, Number 1, pp 1-23, 1999.
+    .. [BVLS] P. B. Start and R. L. Parker, "Bounded-Variable Least-Squares:
+              an Algorithm and Applications", Computational Statistics, 10,
+              129-141, 1995.
 
     Examples
     --------
-    In this example a problem with large sparse matrix and bounds on the
+    In this example the problem with a large sparse matrix and bounds on the
```
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: "a problem" was correct.

File: scipy/optimize/_lsq/lsq_linear.py (hunk 1 above)
Comment (nmayorov) [inline scipy/optimize/_lsq/lsq_linear.py]: The idea was that now the meaning depends on `method`, so I decided not to repeat what is written in `tol` section. Agree that it sounds confusing, so I will probably repeat `tol` section explanation.

File: scipy/optimize/_lsq/lsq_linear.py (hunk 2 above)
Comment (nmayorov) [inline scipy/optimize/_lsq/lsq_linear.py]: Very good)

File: scipy/optimize/_lsq/lsq_linear.py (hunk 3 above)
Comment (nmayorov) [inline scipy/optimize/_lsq/lsq_linear.py]: I meant "iterations after initialization". Is "future" a bad word for that here?

File: scipy/optimize/_lsq/lsq_linear.py (hunk 4 above)
Comment (nmayorov) [inline scipy/optimize/_lsq/lsq_linear.py]: Thanks, I doubted this thing.

File: scipy/optimize/_lsq/lsq_linear.py (hunk 1 above)
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: Or maybe refer to the documentation for `tol`?

File: scipy/optimize/_lsq/lsq_linear.py (hunk 3 above)
Comment (larsmans) [inline scipy/optimize/_lsq/lsq_linear.py]: It's sort of confusing on first reading. "Further" is the right word, I think.

**Comment by nmayorov** [comment]
@larsmans, please take another look now.

**Comment by larsmans** [comment]
LGTM, merging.

**Pull request merged**